linux

```diff
@@ -50,7 +50,53 @@ log_info "Installing Redis ..."
     rm redis-stable.tar.gz && \
     rm -Rf redis-stable
 
-  redis-server --daemonize yes
+  sudo adduser --system --group --no-create-home redis
+  if [ ! -f "/etc/systemd/system/redis-server.service"]; then
+    sudo cat > /etc/systemd/system/redis-server.service <<EOF
+[unit]
+description=redis in-memory data store
+after=network.target
+
+[service]
+user=redis
+group=redis
+execstart=/usr/local/bin/redis-server /etc/redis/redis.conf
+execstop=/usr/local/bin/redis-cli shutdown
+restart=always
+
+[install]
+wantedby=multi-user.target
+EOF
+  fi
+
+  sudo mkdir -p /var/log/redis /var/lib/redis /etc/redis
+  sudo chown redis:redis /var/log/redis /var/lib/redis /etc/redis
+  sudo chmod 755 /var/log/redis /var/lib/redis /etc/redis
+
+  FILE="/etc/redis/redis.conf"
+
+  if [ ! -f "$FILE" ]; then
+    sudo bash -c "cat > $FILE" <<EOF
+bind 0.0.0.0
```

Comment:
Does redis need to bind to 0.0.0.0 here? Can it bind to 127.0.0.1 or something a little less open-to-the-world?

Reply:
Ya totally good catch.